src/generator.rs

```diff
@@ -96,25 +96,33 @@ impl<'a> Generator<'a> {
     }
 
     fn add_chapter(&mut self, ch: &Chapter) -> Result<(), Error> {
-        let rendered = self.render_chapter(ch)
-            .sync()
-            .context("Unable to render template")?;
-
-        let path = ch.path.with_extension("html").display().to_string();
-        let mut content = EpubContent::new(path, rendered.as_bytes()).title(format!("{}", ch));
-
-        let level = ch.number.as_ref().map(|n| n.len() as i32 - 1).unwrap_or(0);
-        content = content.level(level);
-
-        self.builder.add_content(content).sync()?;
-
-        // second pass to actually add the sub-chapters
-        for sub_item in &ch.sub_items {
-            if let BookItem::Chapter(ref sub_ch) = *sub_item {
-                self.add_chapter(sub_ch)?;
+        let rendered = self.render_chapter(ch)?;
+
+        match &ch.path {
```

Comment:
You can get rid of the extra indent level with `ok_or_else()`.

```rust
let internal_path = ch.path.as_ref()
    .ok_or_else(|| failure::err_msg(format!("No content file is found by a path = {:?}", ch.path)))?;
```

Reply:
fixed